worlds/tww/randomizers/Dungeons.py

```diff
@@ -0,0 +1,271 @@
+                item_factory("ET Big Key", world),
+                item_factory(["ET Small Key"] * 3, world),
+                item_factory(["ET Dungeon Map", "ET Compass"], world),
+            )
+
+        if not options.required_bosses or "Wind Temple" in world.boss_reqs.required_dungeons:
+            world.dungeons["Wind Temple"] = make_dungeon(
+                "Wind Temple",
+                item_factory("WT Big Key", world),
+                item_factory(["WT Small Key"] * 2, world),
+                item_factory(["WT Dungeon Map", "WT Compass"], world),
+            )
+
+
+def get_dungeon_item_pool(multiworld: MultiWorld) -> list[Item]:
+    """
+    Retrieve the item pool for all The Wind Waker dungeons in the multiworld.
+
+    :param multiworld: The MultiWorld instance.
+    :return: List of dungeon items across all The Wind Waker dungeons.
+    """
+    return [
+        item for world in multiworld.get_game_worlds("The Wind Waker") for item in get_dungeon_item_pool_player(world)
+    ]
+
+
+def get_dungeon_item_pool_player(world: "TWWWorld") -> list[Item]:
+    """
+    Retrieve the item pool for all dungeons specific to a player.
+
+    :param world: The Wind Waker game world.
+    :return: List of items in the player's dungeons.
+    """
+    return [item for dungeon in world.dungeons.values() for item in dungeon.all_items]
+
+
+def get_unfilled_dungeon_locations(multiworld: MultiWorld) -> list[Location]:
+    """
+    Retrieve all unfilled The Wind Waker dungeon locations in the multiworld.
+
+    :param multiworld: The MultiWorld instance.
+    :return: List of unfilled The Wind Waker dungeon locations.
+    """
+    return [
+        location
+        for world in multiworld.get_game_worlds("The Wind Waker")
+        for location in multiworld.get_locations(world.player)
+        if location.dungeon and not location.item
+    ]
+
+
+def modify_dungeon_location_rules(locations: list[Location], dungeon_specific: set[tuple[int, str]]) -> None:
+    """
+    Modify the rules for The Wind Waker dungeon locations based on specific player-requested constraints.
+
+    :param locations: List of dungeon locations to modify.
+    :param dungeon_specific: Set of dungeon-specific item constraints.
+    """
+    for location in locations:
+        orig_rule = location.item_rule
+        if dungeon_specific:
+            # Restrict dungeon items to be in their own dungeons.
+            dungeon = location.dungeon
+            location.item_rule = lambda item, dungeon=dungeon, orig_rule=orig_rule: (
+                not (item.player, item.name) in dungeon_specific or item.dungeon is dungeon
+            ) and orig_rule(item)
+        else:
+            # Restrict dungeon items to be in any dungeon in the player's local world.
+            player = location.player
+            location.item_rule = lambda item, player=player, orig_rule=orig_rule: (item.player == player) and orig_rule(
+                item
+            )
```

Comment:
This does not work as intended. This makes all dungeon locations only accept local items.

These rules are carried over to the Archipelago's main fill.

Reply:
I think this can be removed entirely? I can't find it in the alttp code this dungeon filling code has been copied from.

Dungeon items set to `any_dungeon` are added to `self.options.local_items` in `generate_early`, so the `fill_dungeons_restrictive` should place items set to `any_dungeon` only within local dungeons.

Reply:
You are correct. I was having much trouble getting this to work initially, and I must've never realized the side effects on all the other dungeon locations.